### ISSUE TITLE
Matchzoo metrics only works on ranking

### ISSUE THREAD

**Issue by yunhenk** [NTMC-Community/MatchZoo#710]
### Describe the Question
this happens when do evaluation
```
  File "/usr/local/lib/python3.6/dist-packages/MatchZoo-2.1.0-py3.6.egg/matchzoo/engine/base_model.py", line 333, in evaluate
    raise ValueError("Matchzoo metrics only works on ranking.")
```
and here is my task, following the comment of Classification class
```python
    classify_task = mz.tasks.Classification()
    classify_task.metrics = ['precision']
```
please help check this problem,thanks

**Comment by uduse** [comment]
So what is your question? Why matchzoo metrics do not work for classification?

**Comment by yunhenk** [comment]
> So what is your question? Why matchzoo metrics do not work for classification?

Yes, what metric should i use for classification? Do i need to write one myself?

**Comment by yunhenk** [comment]
Hi, I checked some source code, here is : BaseModel.evaluate: 
```python
    def evaluate(
        self,
        x: typing.Dict[str, np.ndarray],
        y: np.ndarray,
        batch_size: int = 128
    ) -> typing.Dict[BaseMetric, float]:
        result = dict()
        matchzoo_metrics, keras_metrics = self._separate_metrics()
        y_pred = self.predict(x, batch_size)

        for metric in keras_metrics:
            metric_func = keras.metrics.get(metric)
            result[metric] = K.eval(K.mean(
                metric_func(K.variable(y), K.variable(y_pred))))

        if matchzoo_metrics:
            if not isinstance(self.params['task'], tasks.Ranking):
                raise ValueError("Matchzoo metrics only works on ranking.")
            for metric in matchzoo_metrics:
                result[metric] = self._eval_metric_on_data_frame(
                    metric, x['id_left'], y, y_pred)

        return result
```
As for as i can see, the precision metric is a successor of BaseMetric, ie. a matchzoo_metric, which makes it unable to be used on classification tasks, this seems unreasonable.

**Comment by bwanglzu** [comment]
@yunhenk in this case you should make use of keras metrics, see [here](https://keras.io/metrics/)

**Comment by yunhenk** [comment]
@bwanglzu Thanks, got it. By the way, do you consider adding metrics for classification tasks, since keras metrics removed now. And add one classification task tutorial maybe?

**Comment by bwanglzu** [comment]
@yunhenk We'll add tutorials for classification, need some discussion whether to add precision/recall into metrics,  might confused with ranking metrics..

**Comment by Sjzwind** [comment]
> @yunhenk We'll add tutorials for classification, need some discussion whether to add precision/recall into metrics, might confused with ranking metrics..
> 
> hopefully your questions has been resolved, I'll close this issue for now, if you have future questions please feel free to open another one.

there is no any tutorials about classification until today， three months has gone
